framework/bug-mining/download-issues.pl

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env perl
+#
+#-------------------------------------------------------------------------------
+# Copyright (c) 2014-2018 René Just, Darioush Jalali, and Defects4J contributors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#-------------------------------------------------------------------------------
+
+=pod
+
+=head1 NAME
+
+download-issues.pl -- Collect all issues from the project issue tracker.
```

Comment:
Potential enhancement: 
For Github, the issue URL identifies if whether the matching number of an issue or a pull request. For example:
40,https://github.com/FasterXML/jackson-core/issues/40
39,https://github.com/FasterXML/jackson-core/pull/39
"40" represents an issue, while "39" represents a pull request. We could add an optional argument to filter this file for only issues, removing pull requests identified by the regex. This should be optional, rather than mandatory, as some pull requests also identify issues (GSON has examples of this).